torch/distributed/fsdp/_optim_utils.py

```diff
@@ -476,16 +476,7 @@ def _flatten_optim_state_dict(
                     len(fqns) == 1
                 ), f"use_orig_params is True but there are multiple FQNs, {fqns}."
                 if optim is not None:  # NamedOptimizer or KeyedOptimizer case.
-                    flat_osd_state[key] = copy.deepcopy(
-                        tree_map_only(
-                            torch.Tensor,
-                            # Keep the step state on CPU.
-                            lambda t: t.cpu()
-                            if t.dim() == 0
-                            else torch.zeros(0, device=t.device, dtype=t.dtype),
-                            unflat_osd_state[fqn],
-                        )
-                    )
+                    flat_osd_state[key] = copy.deepcopy(optim.state[param])
```

Comment:
Does this mean the fake init has been already called here?

Reply:
Yes.